designs/disruption-controls.md

```diff
@@ -0,0 +1,118 @@
+    // Users can use "Never" to disable Consolidation.
+    ConsolidateAfter *NillableDuration `json:"consolidateAfter" hash:"ignore"`
+    // ConsolidationPolicy determines how Karpenter will consider nodes
+    // as candidates for Consolidation.
+    // WhenEmpty uses the same behavior as v1alpha5 TTLSecondsAfterEmpty
+    // WhenUnderutilized uses the same behavior as v1alpha5.ConsolidationEnabled: true
+    ConsolidationPolicy string `json:"consolidationPolicy" hash:"ignore"`
+}
+// Budget specifies periods of times where Karpenter will restrict the
+// number of Node Claims that can be terminated at a time.
+// Unless specified, a budget is always active.
+type Budget struct {
+    // MaxUnavailable dictates how many Node Claims owned by this NodePool
+    // can be terminating at once.
+    // This only respects and considers nodes with a DeletionTimestamp set.
+    MaxUnavailable intstr.IntOrString `json:"maxUnavailable" hash:"ignore"`
+    // Crontab specifies when a budget begins being active.
+    // Crontab uses the same syntax as a Cronjob:
+    // "Minute Hour DayOfMonth Month DayOfWeek"
+    // This is required if Duration is set.
+    Crontab *string `json:"crontab,omitempty" hash:"ignore"`
+    // Duration determines how long a Budget is active since each Crontab hit.
+    // This is required if Crontab is set.
+    Duration *metav1.Duration `json:"duration,omitempty" hash:"ignore"`
+}
+```
+
+## Validation/Defaults
+
+For each `Budget`, users must set a non-negative `MaxUnavailable`. Users can disable scale down for a NodePool by setting this to `0`. Users must either omit both `Crontab` and `Duration` or set both of them, since `Crontab` and `Duration` are inherently linked. Omitting these two fields will be equivalent to an always active `Budget`. If a user defines seconds for `Duration`, Karpenter will ignore it, effectively rounding down to the nearest minute, since the smallest denomination of time in upstream Crontabs are minutes.
+
+- An omitted `Budgets` will be defaulted to one `Budget` with `MaxUnavailable: 10%` if left undefined.
+- `ConsolidationPolicy` will be defaulted to `WhenUnderutilized`, with a `consolidateAfter` value of `15s`, which is the same value for Consolidation in v1alpha5.
+
+```
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: default
+spec:
+  disruption:
+    consolidationPolicy: WhenUnderutilized
+    consolidateAfter: 15s
+    budgets:
+    - maxUnavailable: 10%
+```
+
+### Defaulting Alternatives Explored
+
+If Karpenter were to persist a default for the `Crontab` and `Duration` fields, it would need to pick sane values for an “always enabled” Budget. The proposal was to default `Crontab` to `* * * * *` and `Duration` to `1m`, but since `Duration` has no impact outside of the `Crontab`, users who want to set their `Crontab` but forget to set `Duration` may find this a sharp edge.
+
+## Pros and Cons For In-line with the NodePool
```

Comment:
Could you collapse this section into the next section and call that section something like `API Options`. That way all the solutions are enumerated within the same section.

Reply:
Great idea.